alf/algorithms/algorithm.py

```diff
@@ -220,6 +221,7 @@ def _get_opt_and_var_sets(self):
                 else:
                     raise ValueError("Unsupported module type %s" % module)
             opt_and_var_sets.append((opt, vars))
+        self._optimizer_info = '\n\n'.join(optimizer_info)
```

Comment:
This is quite hacky. I suggest call algorithm.get_optimizer_and_module_sets() outside and generate the info there.

Reply:
Moved the optimizer info to a member function.